satpy/readers/abi_base.py

```diff
@@ -0,0 +1,217 @@
+        elif 'goes_lat_lon_projection' in self.nc.keys():
+            return self._get_areadef_latlon(key)
+        else:
+            raise ValueError('Unsupported projection found in the dataset')
+
+    def _get_areadef_latlon(self, key):
+        """Get the area definition of the data at hand.
+        """
+        projection = self.nc["goes_lat_lon_projection"]
+
+        a = projection.attrs['semi_major_axis']
+        b = projection.attrs['semi_minor_axis']
+        fi = projection.attrs['inverse_flattening']
+        pm = projection.attrs['longitude_of_prime_meridian']
+
+        proj_ext = self.nc["geospatial_lat_lon_extent"]
+
+        w_lon = proj_ext.attrs['geospatial_westbound_longitude']
+        e_lon = proj_ext.attrs['geospatial_eastbound_longitude']
+        n_lat = proj_ext.attrs['geospatial_northbound_latitude']
+        s_lat = proj_ext.attrs['geospatial_southbound_latitude']
+
+        lat_0 = proj_ext.attrs['geospatial_lat_center']
+        lon_0 = proj_ext.attrs['geospatial_lon_center']
+
+        area_extent = (w_lon, s_lat, e_lon, n_lat)
+        proj_dict = {'proj': 'latlong',
+                     'lon_0': float(lon_0),
+                     'lat_0': float(lat_0),
+                     'a': float(a),
+                     'b': float(b),
+                     'fi': float(fi),
+                     'pm': float(pm)}
+
+        ll_area_def = geometry.AreaDefinition(
+            self.nc.attrs.get('orbital_slot', 'GOES-R'),
+            self.nc.attrs.get('spatial_resolution', 'ABI L2+ file area'),
+            'abi_l2+_latlon',
+            proj_dict,
+            self.ncols,
+            self.nlines,
+            np.asarray(area_extent))
+
+        return ll_area_def
+
+    def _get_areadef_fixedgrid(self, key):
+        """Get the area definition of the data at hand.
+        """
+        projection = self.nc["goes_imager_projection"]
+
+        a = projection.attrs['semi_major_axis']
+        b = projection.attrs['semi_minor_axis']
+        h = projection.attrs['perspective_point_height']
+
+        lon_0 = projection.attrs['longitude_of_projection_origin']
+        sweep_axis = projection.attrs['sweep_angle_axis'][0]
+
+        # compute x and y extents in m
+        h = float(h)
+
+        x = self.nc['x']
+        y = self.nc['y']
+        x_l = h * x[0]
+        x_r = h * x[-1]
+        y_l = h * y[-1]
+        y_u = h * y[0]
+
+        x_half = (x_r - x_l) / (self.ncols - 1) / 2.
+        y_half = (y_u - y_l) / (self.nlines - 1) / 2.
+        area_extent = (x_l - x_half, y_l - y_half, x_r + x_half, y_u + y_half)
+
+        proj_dict = {'proj': 'geos',
+                     'lon_0': float(lon_0),
+                     'a': float(a),
+                     'b': float(b),
+                     'h': h,
+                     'units': 'm',
+                     'sweep': sweep_axis}
+
+        fg_area_def = geometry.AreaDefinition(
+            self.nc.attrs.get('orbital_slot', 'GOES-R'),  # "GOES-East", "GOES-West"
```

Comment:
In the original abi_l1b.py, this was `'abi_geos'`, does it matter ?

Reply:
Also, how about GOES 17?

Reply:
Yeah I'm not a huge fan of `GOES-R` being mentioned here at all. Either `GOES-East`/`GOES-16` or `GOES-West`/`GOES-17`. I guess the mission is called "GOES-R" though...I guess that's better than `abi_geos` at least.